packages/client/src/publication/Publication.ts

```diff
@@ -91,4 +135,151 @@ export class Publication {
+  > {
+    return execute(this.authentication, async (headers) => {
+      const result = await this.sdk.CreateCommentTypedData(
+        {
+          request,
+          options,
+        },
+        headers,
+      );
+
+      return result.data.result;
+    });
+  }
+
+  async createCommentViaDispatcher(
+    request: CreatePublicCommentRequest,
+  ): PromiseResult<
+    RelayerResultFragment | RelayErrorFragment,
+    CredentialsExpiredError | NotAuthenticatedError
+  > {
+    return execute(this.authentication, async (headers) => {
+      const result = await this.sdk.CreateCommentViaDispatcher({ request }, headers);
+
+      return result.data.result;
+    });
+  }
+
+  async createMirrorTypedData(
+    request: CreateMirrorRequest,
+    options?: TypedDataOptions,
+  ): PromiseResult<
+    InferResultType<CreateMirrorTypedDataMutation>,
+    CredentialsExpiredError | NotAuthenticatedError
+  > {
+    return execute(this.authentication, async (headers) => {
+      const result = await this.sdk.CreateMirrorTypedData(
+        {
+          request,
+          options,
+        },
+        headers,
+      );
+
+      return result.data.result;
+    });
+  }
+
+  async createMirrorViaDispatcher(
+    request: CreateMirrorRequest,
+  ): PromiseResult<
+    RelayerResultFragment | RelayErrorFragment,
+    CredentialsExpiredError | NotAuthenticatedError
+  > {
+    return execute(this.authentication, async (headers) => {
+      const result = await this.sdk.CreateMirrorViaDispatcher({ request }, headers);
+
+      return result.data.result;
+    });
+  }
+
+  async createCollectTypedData(
+    request: CreateCollectRequest,
+    options?: TypedDataOptions,
+  ): PromiseResult<
+    InferResultType<CreateCollectTypedDataMutation>,
+    CredentialsExpiredError | NotAuthenticatedError
+  > {
+    return execute(this.authentication, async (headers) => {
+      const result = await this.sdk.CreateCollectTypedData(
+        {
+          request,
+          options,
+        },
+        headers,
+      );
+
+      return result.data.result;
+    });
+  }
+
+  async createAttachMediaData(
+    request: PublicMediaRequest,
+  ): Promise<InferResultType<CreateAttachMediaDataMutation>> {
+    const result = await this.sdk.CreateAttachMediaData({ request });
+
+    return result.data.result;
+  }
+
+  async hide(
+    request: HidePublicationRequest,
+  ): PromiseResult<void, CredentialsExpiredError | NotAuthenticatedError> {
+    return execute(this.authentication, async (headers) => {
+      await this.sdk.HidePublication({ request }, headers);
+    });
+  }
+
+  async report(
+    request: ReportPublicationRequest,
+  ): PromiseResult<void, CredentialsExpiredError | NotAuthenticatedError> {
+    return execute(this.authentication, async (headers) => {
+      await this.sdk.ReportPublication({ request }, headers);
+    });
+  }
+
+  buildReportReason(reason: PublicationReportReason): ReportingReasonInputParams {
```

Comment:
private?

Reply:
nope, that is another helper, used like

```
publication.report({
    publicationId: '0x014e-0x0a',
    reason: publication.buildReportReason(PublicationReportReason.FAKE_ENGAGEMENT),
    additionalComments: 'comment',
})
```

I created a new ticket to figure out the right way of exposing helpers https://avara.height.app/T-7156
they are also all documented in the TODO file, like

```
- [H] - helper

- [H] client.publication.buildReportReason(PublicationReportReason)
```